src/de.rs

```diff
@@ -1710,10 +1764,8 @@ impl<'a> Deserializer<'a> {
     }
 
     fn error(&self, at: usize, kind: ErrorKind) -> Error {
-        let mut err = Error::from_kind(kind);
-        let (line, col) = self.to_linecol(at);
-        err.inner.line = Some(line);
-        err.inner.col = col;
+        let mut err = Error::from_kind(Some(at), kind);
+        err.fix_linecol(|at| self.to_linecol(at));
```

Comment:
Since `at` is passed in above, isn't this not necessary because it'll never be run?

Reply:
Here, the `at` passed to `Error::from_kind` and the `at` argument in the `fix_linecol` closure, are def the same. But you still need to call `fix_linecol`. Do you mean that you prefer:

```
let mut err = Error::from_kind(Some(at), kind);
err.fix_linecol(|_| self.to_linecol(at));
```

Happy to PR is.

Reply:
Oh nah I think I misunderstood what was happening here, seems fine as-is!